app/assets/javascripts/discourse/app/models/user.js

```diff
@@ -560,6 +560,29 @@ const User = RestModel.extend({
     });
   },
 
+  trustedSession() {
+    return ajax("/u/trusted-session.json");
+  },
+
+  createPasskey() {
+    return ajax("/u/create_passkey.json", {
+      type: "POST",
+    });
+  },
+
+  registerPasskey(credential) {
+    return ajax("/u/register_passkey.json", {
+      data: credential,
+      type: "POST",
+    });
+  },
+
+  deletePasskey(id) {
+    return ajax(`/u/delete_passkey/${id}`, {
+      type: "DELETE",
+    });
+  },
```

Comment:
I know we already did this in the past, but that feels very wrong to have this on user model to me @davidtaylorhq

Reply:
Not particularly attached to this pattern either, but we do this extensively (topic, post, group, category), all have similar calls.

Reply:
Yes I was mostly pointing david as I think we should try to promote a better pattern on this, fine not keep it in this PR.